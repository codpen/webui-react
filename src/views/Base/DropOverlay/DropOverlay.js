import React from "react";

const DropOverlay = (props) => (<div data-test="dropOverlay"
                                     style={{
                                         position: 'absolute',
                                         top: 0,
                                         left: 0,
                                         height: '100%',
                                         width: '100%',
                                         zIndex: 1,
                                         opacity: 0.5,
                                         backgroundColor: 'gray',
                                     }}
/>);

export default DropOverlay;