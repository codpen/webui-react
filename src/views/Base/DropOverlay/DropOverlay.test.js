import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import DropOverlay from "./DropOverlay";
import {findByTestAttr, testStore} from "../../../../Utils";


const setUp = (intialState = {}, props = {}) => {
        const store = testStore(intialState);

        const component = shallow(<DropOverlay {...props} store={store}/>);
        return component;
};


describe('Drop Overlay', function () {
        describe('renders', function () {
                let wrapper;
                beforeEach(() => {
                        const initialState = {};

                        const props = {};
                        wrapper = setUp(initialState, props)
                });

                it('should render without crashing', function () {
                        const component = findByTestAttr(wrapper, "dropOverlay");
                        expect(component).toHaveLength(1);
                });

                it('should match snapshot', function () {
                        expect(toJson(wrapper)).toMatchSnapshot();
                });

        });
});
