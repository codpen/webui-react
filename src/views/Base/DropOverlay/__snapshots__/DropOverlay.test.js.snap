// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Overlay renders should match snapshot 1`] = `
<div
  data-test="dropOverlay"
  style={
    Object {
      "backgroundColor": "gray",
      "height": "100%",
      "left": 0,
      "opacity": 0.5,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 1,
    }
  }
/>
`;
